### PR TITLE
Include micropip from a separate repository

### PR DESCRIPTION
**Pull request by rth**
Closes https://github.com/pyodide/pyodide/issues/3093

Includes micropip from https://github.com/pyodide/micropip

One thing I'm not sure about is whether we still want to run the micropip tests there or if we rely on them passing in https://github.com/pyodide/micropip. Also, it's not very clear to me right now how to run tests from that repo here, particularly since we don't want tests to be included in the wheel.

We should probably have a separate documentation site for micropip, in that repo, so they are updated independently from the main docs.

### CODE REVIEW

**Comment by ryanking13** [comment]
> One thing I'm not sure about is whether we still want to run the micropip tests there or if we rely on them passing in https://github.com/pyodide/micropip.

I think we should run micropip tests in `pyodide/micropip`. It looks like `pyodide/micropip` vendors subset of `src/py` to run tests locally (outside of browser), but I think it is okay to drop outside of browser support and run full test suite inside Pyodide.

> We should probably have a separate documentation site for micropip, in that repo, so they are updated independently from the main docs.

Yes, indeed we need a seperate docs for micropip.

**Comment by rth** [comment]
> outside of browser support and run full test suite inside Pyodide.

I'm not sure you can do that, because a number of tests need to start an HTTP server which you can't do from within the browser (e.g. [here](https://github.com/pyodide/micropip/blob/main/tests/test_micropip.py#L284)). It's also rather practical to have host tests when possible, as it for instance is correctly counted toward code coverage.

Also, it's a bit questionable whether it's useful to run those for each commit in this repo, as opposed to when updating micropip only (though yes, this would be harder to setup). 

The other solution is we could keep some of the tests in `test_micropip.py`, and have a more extensive test suite in the other repo.

**Comment by ryanking13** [comment]
> The other solution is we could keep some of the tests in `test_micropip.py`, and have a more extensive test suite in the other repo.

That sounds reasonable to me.

**Comment by rth** [comment]
CI is good, can anyone have a look so we can merge this? Thanks!

File: packages/micropip/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+
+source:
+  sha256: 693a7fd7ce92fd95eaf9d88fe243f45951ad5c694cccf89493cbcc3666bbdeab
+  url: https://files.pythonhosted.org/packages/14/19/e5d36b2cc156c66858493dc118956f6fed2b1408e62be12e4ffed392b7cb/micropip-0.1.0-py3-none-any.whl
+
 requirements:
   run:
     - pyparsing
     - packaging
-    - distutils # TODO: remove once there is a release with https://github.com/pypa/packaging/pull/396
```
Comment (ryanking13) [inline packages/micropip/meta.yaml:11]: It is good to see micropip having fewer dependencies!

**Review by ryanking13 — APPROVED** [review]
Thanks a lot @rth!

**Pull request merged**